lib/godmin/engine_wrapper.rb

```diff
@@ -2,32 +2,47 @@ module Godmin
   class EngineWrapper
     attr_reader :engine
 
-    def initialize(obj)
-      @engine = find_engine(obj)
+    def initialize(controller)
+      @engine = find_engine(controller)
     end
 
-    def root
-      engine.root
+    def namespace
+      @namespace ||= engine.railtie_namespace
     end
 
-    def namespace
-      engine.railtie_namespace.to_s.underscore
+    def namespaced?
+      @namespaced ||= namespace.present?
+    end
+
+    def namespaced_path
```

Comment:
This is how the rails methods are named, e.g. https://github.com/rails/rails/blob/848776c3da0e4c45d77ad573bb2a250845522b96/railties/lib/rails/generators/named_base.rb#L97